mypy/typestate.py

```diff
@@ -3,22 +3,26 @@
 and potentially other mutable TypeInfo state. This module contains mutable global state.
 """
 
-from typing import Dict, Set, Tuple
+from typing import Dict, Set, Tuple, Optional
 
 MYPY = False
 if MYPY:
     from typing import ClassVar
 from mypy.nodes import TypeInfo
 from mypy.types import Instance
+from mypy.server.trigger import make_trigger
 
 
 class TypeState:
     """
     This class provides subtype caching to improve performance of subtype checks.
+    As well it holds protocol fine grained dependencies.
```

Comment:
Style nit: There should be an empty line after the first line of the docstring (also we tend to write it as `"""<first line>` (no newline after `"""`).

Grammar nit: Would be better to write as "It also holds ...".